src/main/scala/com/databricks/spark/csv/CsvRelation.scala

```diff
@@ -136,6 +137,29 @@ case class CsvRelation protected[spark] (
     }
   }
 
+  // The function below was borrowed from JSONRelation
+  override def insert(data: DataFrame, overwrite: Boolean) = {
+    val filesystemPath = new Path(location)
+    val fs = filesystemPath.getFileSystem(sqlContext.sparkContext.hadoopConfiguration)
+
+    if (overwrite) {
+      try {
+        fs.delete(filesystemPath, true)
+      } catch {
+        case e: IOException =>
```

Comment:
Shouldn't we fail if this exception happens? If we catch the exception and continue with saving the data, wouldn't we end up with inconsistent data?